### DISCUSSION TITLE
Formatted HTML/JSON in execution logs

### DISCUSSION THREAD

**Discussion by ARLOthePEAR** [burkybang/SuiteAdvanced-Feedback#44] [Q&A] [answered]
Hi! Love your extension, but whenever it's enabled, it disables the other "NetSuite: HTML Script Notes" chrome extension I use to neatly auto-format JSON that appears in script execution logs. Any plans to add this handy feature to your extension?!

Thank you!

**Comment by burkybang** [comment, marked as answer]
Actually, this is already a feature of SuiteAdvanced. Just disable or uninstall the other extension. SuiteAdvanced also takes it a step further with syntax theming options in settings. It also formats JSON/XML that exists in long text fields.

**Reply by ARLOthePEAR** [reply]
Sounds amazing. Unfortunately, it still doesn't seem to work for me in execution logs with SuiteAdvanced enabled, even after deleting the other extension 😕. I do see that it formats JSON nicely in long text fields. Any other tips to get it working in the execution logs?

**Reply by burkybang** [reply]
It only works if it's valid JSON or XML. Can you verify that what you're logging is valid?

**Reply by ARLOthePEAR** [reply]
Ah, I've got it working now, thanks for the help!

**Reply by burkybang** [reply]
Awesome! No prob! 🎉

**Reply by eduardRsult** [reply]
The part that us missing is the "Links to records." feature of that other plugin

**Reply by HeavenlyEntity** [reply]
To add to this. @burkybang if you try to log an entire xml doc string the entire extension freezes up my browser. I have to disable it. It would be nice to toggle according to type either its the literal code or formatted document somehow in the SuiteAdvanced menu. So for example: XML: `['prettier code', 'parsed']` HTML: `['prettier code', 'parsed']`, ect. In a dropdown as options. I think it would be awesome so we can test both the render and the code. I would apply this to only markup languages though.

**Reply by burkybang** [reply]
@HeavenlyEntity It freezes?! Wow! There's a max of 4000 characters for each execution log. Are you able to provide a sample that freezes so I can test?

**Reply by burkybang** [reply]
@eduardRsult The "Links to records" feature is now available. It also supports object scriptids.

![image](https://github.com/burkybang/SuiteAdvanced-Feedback/assets/927886/b0babe8c-e226-40cd-ac62-806b69da8cba)

**Reply by eduardRsult** [reply]
@burkybang This works well for numbers. Thank you very much!! this is my favorite wat of logging records. Well done!

Something i found:
It works halfway for strings of numbers (which occur easily in NS)

when logging   log.debug('pickTask updated', { picktask: "79845" });

your plugin now displays the link as clickable but makes the link
`/app/accounting/wms/picktask/picktask.nl?id="79847"`
 (which does not work) instead of
/app/accounting/wms/picktask/picktask.nl?id=79845

**Reply by burkybang** [reply]
Ah crap. Will fix that. Thanks for letting me know.

**Reply by burkybang** [reply]
It'll be fixed in the next release

![image](https://github.com/burkybang/SuiteAdvanced-Feedback/assets/927886/30a51951-4235-44bd-93e0-63291cf4642a)
